crates/matrix-sdk-crypto/src/identities/manager.rs

```diff
@@ -532,6 +532,9 @@ impl IdentityManager {
                                 if result.any_cleared() {
                                     changed_identity = Some((*private_identity).clone());
                                     info!(cleared = ?result, "Removed some or all of our private cross signing keys");
+                                } else if new && private_identity.has_master_key().await {
```

Comment:
I first thought that you should change this to be a separate if branch instead of an else if. Because, while`if result.any_cleared()` might be true, this does not mean that the master key got rotated.

As long as we have the private part of the master key, and the master key signed the subkeys we could mark the identity as verified.

But since you're also requiring this to be a new identity, i.e. not a changed one, this makes little sense. If the identity is new then there's very little chance that you rotated the subkeys only. So let's stay prudent and only mark things as verified if nothing got rotated.

In any case, let's spell out what's happening here and why it's ok to mark things as verified. Something like:

> If the master key didn't rotate up above, then this means that the public part and the private parts of the master key match. We previously did a signature check, so this means that the private part of the master key has signed the identity. We can safely mark the public part of the identity as verified.

Reply:
Added